hack/ci-notify/main.go

```diff
@@ -0,0 +1,184 @@
+package main
+
+/*
+
+A list of available environment variables can be found here:
+
+https://github.com/kubernetes/test-infra/blob/a9f967cb1235916fb5a4eca661fa083413211242/prow/jobs.md#job-environment-variables
+
+For periodics:
+- JOB_NAME
+- JOB_TYPE
+- JOB_SPEC
+- BUILD_ID
+- PROW_JOB_ID
+
+*/
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/google/go-github/github"
+	"golang.org/x/oauth2"
+)
+
+var (
+	jobName   = flag.String("job-name", "", "Name of the CI job  e.g.  periodic-ci-azure-vmimage")
+	comment   = flag.String("comment", "", "The comment to add.")
+	owner     = flag.String("owner", "openshift", "Name of github organization.")
+	repo      = flag.String("repo", "openshift-azure", "Name of github repository")
+	creator   = flag.String("creator", "openshift-azure-robot", "Github user who will send the notifications.")
+	duration  = flag.Duration("since", 504*time.Hour, "How many hours in the past to be considered during the search since the last issue.")
+	succeeded = flag.Bool("success", false, "Did the ci job succeed?")
+)
+
+type client struct {
+	gh *github.Client
+}
+
+func newClient(token string) *client {
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: token},
+	)
+	tc := oauth2.NewClient(context.Background(), ts)
+	return &client{
+		gh: github.NewClient(tc),
+	}
+}
+
+// listIssues lists github issues that are open and were created by creator. By default
+// it searches 3 weeks into the past.
+func (c *client) listIssues(ctx context.Context, owner, repo string, since time.Duration) ([]*github.Issue, error) {
+	opt := &github.IssueListByRepoOptions{
+		State:   "open",
+		Creator: *creator,
+		Since:   time.Now().Add(-since),
+	}
+
+	issues, _, err := c.gh.Issues.ListByRepo(ctx, owner, repo, opt)
+	if err != nil {
+		return nil, err
+	}
+
+	return issues, nil
+}
+
+func (c *client) getVMImageBuildIssue(ctx context.Context, jobName, owner, repo string, since time.Duration) (*github.Issue, error) {
```

Comment:
The name is very specific. I think we will be using this script not only for `vmimage` builds (`vmimage-validate`, for example). Maybe`getIssueByName` or something like that?